pymc/ode/utils.py

```diff
@@ -103,10 +103,17 @@ def augment_system(ode_func, n_states, n_theta):
 
     # Get symbolic representation of the ODEs by passing tensors for y, t and theta
     yhat = ode_func(t_y, t_t, t_p[n_states:])
-    # Stack the results of the ode_func into a single tensor variable
-    if not isinstance(yhat, (list, tuple)):
-        yhat = (yhat,)
-    t_yhat = at.stack(yhat, axis=0)
+    if isinstance(yhat, at.TensorVariable):
+        t_yhat = at.atleast_1d(yhat)
+    else:
+        # Stack the results of the ode_func into a single tensor variable
+        if not isinstance(yhat, (list, tuple)):
+            yhat = (yhat,)
```

Comment:
The CI says this is not covered by the tests?
I think this line could become an exception: Unexpected type returned by the ODE function.
And you could test it by returning a dictionary or set of ODEs instead of a tensor/list/tuple.

FYI you can generate the coverage report locally by passing `--cov=pymc --cov-report=term-missing`. Needs "pip install pytest-cov" and can be combined with running just a specific test file.

Reply:
I was away over the weekend, but I made the changes that you suggested just now. Hopefully the CI still works.